system/jhtree/ctfile.cpp

```diff
@@ -284,7 +284,7 @@ void CWriteNodeBase::writeHdr()
 
 void CWriteNodeBase::write(IFileIOStream *out, CRC32 *crc)
 {
-    if (isLeaf() && (keyHdr->getKeyType() & HTREE_COMPRESSED_KEY))
+    if (isBlob() || (isLeaf() && (keyHdr->getKeyType() & HTREE_COMPRESSED_KEY)))
```

Comment:
I think this is the fix for the corrupted info on writing blobs in a non-compressed index